src/main/java/seedu/address/model/property/PropertyPredicateList.java

```diff
@@ -0,0 +1,33 @@
+package seedu.address.model.property;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * List of {@code Predicate<Property>}, supports equality comparisons
+ * with other {@code PropertyPredicateLists}, and combining of all
+ * predicates it holds into a single predicate.
+ */
+public class PropertyPredicateList {
+    private List<Predicate<Property>> predicates;
+
+    public PropertyPredicateList(List<Predicate<Property>> predicates) {
+        this.predicates = predicates;
+    }
+
+    /**
+     * Combines all predicates in its internal list into a single
+     * predicate.
+     * @return {@code Predicate<Property>} that is combined with and
```

Comment:
is there supposed to be something else after the and?

Reply:
The `and` here is referring to the logical AND, I think I made it quite ambiguous here. Will go change it.